pymc/printing.py

```diff
@@ -26,91 +28,61 @@
         )
-        if include_params:
-            if print_name:
-                return r"${} \sim {}({})$".format(
-                    print_name, op_name, ",~".join([d.strip("$") for d in dist_args])
-                )
-            else:
-                return r"${}({})$".format(op_name, ",~".join([d.strip("$") for d in dist_args]))
-
-        else:
-            if print_name:
-                return rf"${print_name} \sim {op_name}$"
-            else:
-                return rf"${op_name}$"
-
-    else:  # plain
-        dist_name = (
-            dist.owner.op._print_name[0] if hasattr(dist.owner.op, "_print_name") else "Unknown"
+    var_name, dist_name, args_str = _get_varname_distname_args(
+        var, formatting=formatting, dist_name=dist_name
+    )
+    if "include_params" in kwargs:
+        warnings.warn(
+            "The `include_params` argument has been deprecated. Future versions will always include parameter values.",
+            FutureWarning,
+            stacklevel=2,
         )
-        if include_params:
-            if print_name:
-                return r"{} ~ {}({})".format(print_name, dist_name, ", ".join(dist_args))
-            else:
-                return r"{}({})".format(dist_name, ", ".join(dist_args))
-        else:
-            if print_name:
-                return rf"{print_name} ~ {dist_name}"
-            else:
-                return dist_name
+        # When removing, remove **kwargs from here and str_for_model().
+        if not kwargs["include_params"]:
+            args_str = ""
+    if formatting == "latex":
+        out = rf"${var_name} \sim {dist_name}({args_str})$"
+    elif formatting == "plain":
+        var_name = var_name.replace("~", "-")
+        dist_name = dist_name.replace("~", "-")
+        args_str = args_str.replace("~", "-")
+        out = f"{var_name} ~ {dist_name}({args_str})"
+    else:
+        raise ValueError(
+            f"Formatting method not recognized: {formatting}. Supported formatting: [latex, plain]."
+        )
+    return out
 
 
-def str_for_model(model: Model, formatting: str = "plain", include_params: bool = True) -> str:
-    """Make a human-readable string representation of Model, listing all random variables
-    and their distributions, optionally including parameter values."""
+def str_for_model(model: Model, formatting: str = "plain", **kwargs) -> str:
+    """Make a human-readable string representation of Model including all random
+    variables and their distributions.
+    """
     all_rv = itertools.chain(model.unobserved_RVs, model.observed_RVs, model.potentials)
-
-    rv_reprs = [rv.str_repr(formatting=formatting, include_params=include_params) for rv in all_rv]
-    rv_reprs = [rv_repr for rv_repr in rv_reprs if "TransformedDistribution()" not in rv_repr]
-
+    rv_reprs = [rv.str_repr(formatting=formatting, **kwargs) for rv in all_rv]
```

Comment:
I know it's not the point of this PR but can we already remove the assumption that model RVs have this `str_repr` attached to it? 

Just call the respective function `str_for_dist` or `str_for_deterministic` on the RV

It would fix https://github.com/pymc-devs/pymc/issues/6311

Reply:
Interesting, I hadn't been aware of that issue. I don't think that we can quite yet remove that assumption, essentially because right now the str_repr function (`str_for_model_var` in this PR, or `str_for_dist` and `str_for_deterministic` in mainline) doesn't know how to distinguish between deterministics and potentials.

At the moment that info is inserted using `functools.partial` whenever the Model registers a new deterministic or potential. But, related to the above comment: currently we don't require that a model object is on the context stack in order to pretty-print a model variable. Requiring a model object would simplify this though

Reply:
Yes it would be fine to require a model to be on the context for the fancier pretty printing.

We can have a much more limited pretty printing for standalone variables (or none at all)